config/runtimes/vllm-multinode-template.yaml

```diff
@@ -0,0 +1,259 @@
+                  if [[ -z $gpu_status ]]; then
+                    echo "$1: GPU does not exist"
+                    exit 1
+                  fi
+
+                  used_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f1)
+                  reserved_gpu=$(echo "$gpu_status" | awk '{print $1}' | cut -d'/' -f2)
+
+                  # Determine health status based on GPU usage
+                  if [[ "$used_gpu" != "$reserved_gpu" ]]; then
+                    echo "$1: Unhealthy - Used: $used_gpu, Reserved: $reserved_gpu"
+                    exit 1
+                  fi
+                  
+                  # Check model health
+                  curl --silent --max-time 5 --fail-with-body http://localhost:8080/health
+          startupProbe:
+            failureThreshold: 40
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 5
+            exec:
+              command:
+                - bash
+                - -c
+                - |
+                  ray status
+
+                  # Check model health
+                  curl --silent --max-time 5 --fail-with-body http://localhost:8080/health
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+      volumes:
+        - name: shm
+          emptyDir:
+            medium: Memory
+            sizeLimit: 12Gi    
+      workerSpec:
+        pipelineParallelSize: 2
+        tensorParallelSize: 1
+        containers:
+          - name: worker-container
+            image: $(vllm-image)
+            command: ["bash", "-c"]
+            args:
+              - |
+                SECONDS=0
+
+                while true; do              
+                  if (( SECONDS <= 120 )); then
+                    if ray health-check --address "${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379" > /dev/null 2>&1; then
+                      echo "Global Control Service(GCS) is ready."
+                      break
+                    fi
+                    echo "$SECONDS seconds elapsed: Waiting for Global Control Service(GCS) to be ready."
+                  else
+                    if ray health-check --address "${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379"; then
+                      echo "Global Control Service(GCS) is ready. Any error messages above can be safely ignored."
+                      break
+                    fi
+                    echo "$SECONDS seconds elapsed: Still waiting for Global Control Service(GCS) to be ready."
+                    echo "For troubleshooting, refer to the FAQ at https://docs.ray.io/en/master/cluster/kubernetes/troubleshooting/troubleshooting.html#kuberay-troubleshootin-guides"
+                  fi
+                  
+                  sleep 5
+                done
+
+                echo "Attempting to connect to Ray cluster at $RAY_HEAD_ADDRESS ..."
+                RAY_HEAD_ADDRESS="${HEAD_SVC}.${POD_NAMESPACE}.svc.cluster.local:6379"
```

Comment:
Perhaps RAY_HEAD_ADDRESS should be set before the `echo`?

Reply:
good catch!. updated.